docs/source/markdown/podman-build.1.md

```diff
@@ -438,10 +438,14 @@ When the option is specified or set to "true", pull the image from the first
 registry it is found in as listed in registries.conf.  Raise an error if not
 found in the registries, even if the image is present locally.
 
-If the option is disabled (with *--pull=false*), or not specified, pull the
+If the option is disabled (with *--pull=false*), pull the
 image from the registry only if the image is not present locally. Raise an
 error if the image is not found in the registries.
 
+If the option is not specified, pull the image from the registry if it is not
+present locally or if there is a newer version on the registry.  Raise an
```

Comment:
This is technically incorrect.  It pulls if the registry version is different, not necessarily newer.

Image I create a container image and push it to a registry as :latest, later I find there is a big bug in my version, and remove it, causing the previous version to be :latest.  Now if someone had pulled in my buggy image, and later pulls again they will revert back to the older image, even though they were PullIfNewer.

Container/Image has no way to determine whether or not one image is "NEWER" then another image, it can only check if the digest is different.  We recently received a bug on this misunderstanding from a user, so we should not be documenting it incorrectly in the man pages and other docs.

Reply:
@vrothberg I thought you had recently changed the check to pull only if the image had a newer date stamp, not that it had changed.  Am I misremembering?

Reply:
I don't know, but as @mtrmac pointed out in a different issue/pr this is a bad idea, since it will not remove a bad image unless someone pushes a "new" version.  As opposed to removing the old "new" image.

Reply:
> @vrothberg I thought you had recently changed the check to pull only if the image had a newer date stamp, not that it had changed. Am I misremembering?

An earlier version of the shortnames PR for Buildah had that but I reverted that part based on the feedback. A digest check is really the way forward. Some images claim to be have been born before me which I doubt :^)